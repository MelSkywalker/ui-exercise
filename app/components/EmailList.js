import React from "react";
import { useSelector } from "react-redux";
import { emailSelector } from '../redux/selectors';
import EmailItem from "./EmailItem";

const EmailList = () => {
  const { emailsById, emailsIdArray } = useSelector(emailSelector);
  return (
    <div >
      {emailsIdArray.map(id => {
        return (
          <EmailItem
            key={id}
            content={emailsById[id]}
          />
        );
      })}
    </div>
  )
};

export default EmailList;