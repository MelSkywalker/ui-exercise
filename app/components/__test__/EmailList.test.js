import React from 'react';
import * as reactRedux from 'react-redux';
import { shallow } from 'enzyme';

import EmailList from '../EmailList';

describe('components | EmailList', () => {
	const useSelectorMock = jest.spyOn(reactRedux, 'useSelector');

	const mockEmailList = {
		emailList: [
			{
				id: "1",
				subject: "hey!",
				tags: ['work', 'travel']
			},
			{
				id: "2",
				subject: "Attention!",
				tags: []
			}
		]
	};

	beforeEach(() => {
		useSelectorMock.mockClear();
	});

	it('renders the component', () => {
		useSelectorMock.mockReturnValue(mockEmailList);
		const component = shallow(<EmailList />);
		expect(component).toMatchSnapshot();
	});
});
