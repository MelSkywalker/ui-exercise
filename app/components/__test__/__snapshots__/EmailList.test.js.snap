// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | EmailList renders the component 1`] = `
<div
  className="email-list"
>
  <EmailItem
    content={
      Object {
        "id": "1",
        "subject": "hello!",
      }
    }
    key="1"
  />
  <EmailItem
    content={
      Object {
        "id": "2",
        "subject": "hello!!",
      }
    }
    key="2"
  />
</div>
`;
